src/data/thunks.js

```diff
@@ -120,15 +121,39 @@ export function startExam() {
 export function startProctoringExam() {
   return async (dispatch, getState) => {
     const { exam } = getState().examState;
+    const { attempt } = exam || {};
     if (!exam.id) {
       logError('Failed to start exam. No exam id.');
       return;
     }
-    await updateAttemptAfter(
-      exam.course_id, exam.content_id, createExamAttempt(exam.id, false, true),
-    )(dispatch);
-    const proctoringSettings = await fetchProctoringSettings(exam.id);
-    dispatch(setProctoringSettings({ proctoringSettings }));
+    const { desktop_application_js_url: workerUrl } = attempt || {};
+    const readyToStart = attempt && attempt.attempt_status === ExamStatus.READY_TO_START;
+    const useWorker = window.Worker && workerUrl;
+
+    // This check if we are currently on the ready_to_start instructions page, if we are
```

Comment:
no, you're right, when I needed to add "start proctored exam" functionality I noticed that we already had a function startProctoredExam, but it only created attempt and I thought that I should just extend it to be able to start exam also
I think I just got confused by the naming since it was already called startProctoredExam, I should've just renamed it to createAttempt and added new function to handle actual starting of the exam
refactored in 64e2f0e

also thanks to this I noticed that we do not need JS worker block in startExam function and removed it since that function is only used for timed exams so we never have to use worker there